_posts/2019-07-23-hibernate-orm-config-profiles.adoc

```diff
@@ -0,0 +1,76 @@
+There are a few common approaches to make the best of it.
+
+The first choice is to use `quarkus.hibernate-orm.database.generation=drop-and-create` in conjunction with `import.sql`.
+That way for every change to your app and in particular to your entities, the database schema will be properly recreated
+and your data fixture (stored in `import.sql`) will be used to repopulate it from scratch.
+This is best to perfectly control your environment.
+
+[NOTE]
+--
+By default, Hibernate ORM, upon boot, will read and execute the SQL statements in the `/import.sql` file (if present).
+You can change the file name by changing the property `quarkus.hibernate-orm.sql-load-script` in `application.properties`.
+--
+
+The second approach is to use `quarkus.hibernate-orm.database.generation=update`.
+This approach is best when you do many entity changes but
+still need to work on a copy of the production data
+or if you want to reproduce a bug that is based on specific database entries.
+`update` is a best effort from Hibernate ORM and will fail in specific situations
+including altering your database structure which could lead to data loss.
+For example if you change structures related to the foreign key, Hibernate ORM might have to bail out.
+But for development, these limitations are acceptable.
+
+The third approach is to use `quarkus.hibernate-orm.database.generation=none`.
+This approach is best when you are working on a copy of the production data but want to fully control the schema evolution.
+Or if you use a database schema migration tool like https://quarkus.io/guides/flyway-guide[Flyway].
+When you make a change to an entity, make sure to manually adapt the database schema accordingly.
+
+WARNING: Do not use `quarkus.hibernate-orm.database.generation` `drop-and-create` and `update` in your production environment. You have been warned :)
+
+== Now with Quarkus profiles
+
+It becomes really powerful when combined with Quarkus configuration profiles.
+You can define different https://quarkus.io/guides/application-configuration-guide#configuration-profiles[configuration profiles]
+to select different behaviors depending on your environment.
+This is great because you can define different combinations of Hibernate ORM properties depending matching the development style you currently need.
+
+[source]
+.application.properties
+--
+# By default, use the clean (data) slate approach
+%dev.quarkus.hibernate-orm.database.generation = drop-and-create
+%dev.quarkus.hibernate-orm.sql-load-script = import-dev.sql
+
+# Use this to incrementally work in your app while keeping data
+# Useful for rapid dev-check cycles with Quarkus dev mode
+%dev-with-data.quarkus.hibernate-orm.database.generation = update
+%dev-with-data.quarkus.hibernate-orm.sql-load-script =
+
+# LET's make sure we don't wipe the production data by accident!
+%prod.quarkus.hibernate-orm.database.generation = none
+%prod.quarkus.hibernate-orm.sql-load-script =
```

Comment:
Are we clear that it does work only if you don't have an `import.sql`. It might be a bit surprising for people.

Reply:
I had forgotten about that. But I knew when I wrote this pattern.
I wonder how we can improve the situation. Right now we raise an exception when the file is not present in `HibernateOrmProcessor:445`.
We could stop doing that and let typos go through.

Or we could document that an explicit value `no-file` means disabling the file loading.
We need to check if underneath Hibernate ORM complains of that fake file though

Reply:
Why not simply omit this line?

It should be up to the users if they want to import a script; even while I disagree it shouldn't be encouraged, there are valid use cases even for production.